### PR TITLE
Include the offline report in an archive

### PR DESCRIPTION
**Pull request by hughsie**
This allows us to upload *one* file to the LVFS when including a signature.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — COMMENTED** [review]
If making a single archive for this purpose - why not just send up the key to verify it in the archive too?

**Comment by hughsie** [comment]
> why not just send up the key to verify it in the archive too

I'm not sure what it adds to be honest -- if we're just validating that a given certificate generated a signature that can be sidestepped with 2 `certtool` commands... If we're using `--sign` then I think the user should have uploaded the cert.

**Comment by superm1** [comment]
> > why not just send up the key to verify it in the archive too
> 
> I'm not sure what it adds to be honest -- if we're just validating that a given certificate generated a signature that can be sidestepped with 2 `certtool` commands... If we're using `--sign` then I think the user should have uploaded the cert.

I guess that's true.

File: src/fu-util.c [hunk 1]
```diff
@@ -1686,14 +1688,23 @@ fu_util_report_export(FuUtilPrivate *priv, gchar **values, GError **error)
...
+			sig_img = fu_firmware_new_from_bytes(sig_blob);
+			fu_firmware_set_id(sig_img, "report.json.p7c");
+			fu_firmware_add_image(archive, sig_img);
 		}
+
+		/* save to local file */
+		fu_archive_firmware_set_format(FU_ARCHIVE_FIRMWARE(archive), FU_ARCHIVE_FORMAT_ZIP);
+		fu_archive_firmware_set_compression(FU_ARCHIVE_FIRMWARE(archive),
+						    FU_ARCHIVE_COMPRESSION_GZIP);
+		filename = g_strdup_printf("%s.zip", fwupd_device_get_id(dev));
```
Comment (superm1) [inline src/fu-util.c:1701]: Shouldn't this be .gz since it's gzip?

Comment (hughsie) [reply src/fu-util.c:1701]: Nah, it's a bit of a weird misnomer -- it's a ZIP container format with MSZIP block compression. Maybe it makes sense to call the archive something else -- e.g. a `a45df35ac0e948ee180fe216a5f703f32dda163f.rzip` file looks less inviting to open and edit than a `a45df35ac0e948ee180fe216a5f703f32dda163f.zip` file..

Comment (superm1) [reply src/fu-util.c:1701]: Sounds good

Comment (hughsie) [reply src/fu-util.c]: Done. I used `.fwupdreport`

**Review by superm1 — APPROVED** [review]

**Pull request merged**